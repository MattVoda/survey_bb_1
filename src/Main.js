import React, { Component } from "react";
import SurveyPage from "./SurveyPage";
import ResultsPage from "./ResultsPage";

export default class Main extends Component {
  constructor(props) {
    super(props); //black magic
    this.state = {
      surveyStage: "0",
      industry: "",
      employeeCount: "",
      SliderInput1Value: "",
      RadioInput1Value: ""
    };
    //this.handleChange = this.handleChange.bind(this);
    this.handleStageSelection = this.handleStageSelection.bind(this);
  }

  handleStageSelection(surveyStage) {
    this.setState({
      surveyStage: surveyStage
    });
  }

  // look up spread {{...state}}
  render() {
    return (
      <div>
        <SurveyPage
          surveyStage={this.state.surveyStage}
          industry={this.state.industry}
          employeeCount={this.state.employeeCount}
          SliderInput1Value={this.state.SliderInput1Value}
          RadioInput1Value={this.state.RadioInput1Value}
          onStageSelect={this.handleStageSelection}
        />
        <ResultsPage
          industry={this.state.industry}
          employeeCount={this.state.employeeCount}
          SliderInput1Value={this.state.SliderInput1Value}
          RadioInput1Value={this.state.RadioInput1Value}
        />
      </div>
    );
  }
}
