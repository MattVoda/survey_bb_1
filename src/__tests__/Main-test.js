import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Main from '../Main';

test("radio button advances progress bar", () => {
  const component = shallow(
    <Main />
  );
  
  expect(component.html()).toMatchSnapshot();
});