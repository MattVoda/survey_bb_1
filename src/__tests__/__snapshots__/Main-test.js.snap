exports[`test radio button advances progress bar 1`] = `"<div><div style=\"margin:100px;\"><div class=\"ant-steps ant-steps-horizontal ant-steps-label-horizontal ant-steps-hidden\"><div class=\"ant-steps-item ant-steps-status-process ant-steps-custom\" style=\"margin-right:-1px;\"><div class=\"ant-steps-tail\" style=\"padding-right:1px;\"><i></i></div><div class=\"ant-steps-step\"><div class=\"ant-steps-head\"><div class=\"ant-steps-head-inner\"><span class=\"ant-steps-icon\"><i class=\"anticon anticon-user\"></i></span></div></div><div class=\"ant-steps-main\"><div class=\"ant-steps-title\">People</div></div></div></div><div class=\"ant-steps-item ant-steps-status-wait ant-steps-custom\" style=\"margin-right:-1px;\"><div class=\"ant-steps-tail\" style=\"padding-right:1px;\"><i></i></div><div class=\"ant-steps-step\"><div class=\"ant-steps-head\"><div class=\"ant-steps-head-inner\"><span class=\"ant-steps-icon\"><i class=\"anticon anticon-solution\"></i></span></div></div><div class=\"ant-steps-main\"><div class=\"ant-steps-title\">Process</div></div></div></div><div class=\"ant-steps-item ant-steps-status-wait ant-steps-custom\" style=\"margin-right:-1px;\"><div class=\"ant-steps-tail\" style=\"padding-right:1px;\"><i></i></div><div class=\"ant-steps-step\"><div class=\"ant-steps-head\"><div class=\"ant-steps-head-inner\"><span class=\"ant-steps-icon\"><i class=\"anticon anticon-credit-card\"></i></span></div></div><div class=\"ant-steps-main\"><div class=\"ant-steps-title\">Technology</div></div></div></div><div class=\"ant-steps-item ant-steps-item-last ant-steps-status-wait ant-steps-custom\"><div class=\"ant-steps-step\"><div class=\"ant-steps-head\"><div class=\"ant-steps-head-inner\"><span class=\"ant-steps-icon\"><i class=\"anticon anticon-smile-o\"></i></span></div></div><div class=\"ant-steps-main\"><div class=\"ant-steps-title\">Results</div></div></div></div></div><div><hr/><br/><br/><div class=\"ant-row\"><div class=\"ant-col-10\"><h3>Number of employees: </h3></div><div class=\"ant-col-2\"></div><div class=\"ant-col-12\"><div class=\"ant-input-number\"><div class=\"ant-input-number-handler-wrap\"><span unselectable=\"unselectable\" class=\"ant-input-number-handler ant-input-number-handler-up \"><span unselectable=\"unselectable\" class=\"ant-input-number-handler-up-inner\"></span></span><span unselectable=\"unselectable\" class=\"ant-input-number-handler ant-input-number-handler-down \"><span unselectable=\"unselectable\" class=\"ant-input-number-handler-down-inner\"></span></span></div><div class=\"ant-input-number-input-wrap\"><input min=\"1\" max=\"100000\" class=\"ant-input-number-input\" autocomplete=\"off\" value=\"1000\"/></div></div></div></div><br/><br/><div class=\"ant-row\"><div class=\"ant-col-10\"><h3>Industry: </h3></div><div class=\"ant-col-2\"></div><div class=\"ant-col-12\"><a class=\"ant-dropdown-link ant-dropdown-trigger\" href=\"#\">Select Industry <i class=\"anticon anticon-down\"></i></a></div></div><br/><br/><div class=\"ant-row\"><div class=\"ant-col-10\"><h3>Agile teams are empowered at your company: </h3></div><div class=\"ant-col-2\"></div><div class=\"ant-col-12\"><div class=\"ant-slider ant-slider-with-marks\"><div class=\"ant-slider-rail\"></div><div class=\"ant-slider-track\" style=\"visibility:visible;left:0%;width:0%;\"></div><div class=\"ant-slider-step\"><span class=\"ant-slider-dot ant-slider-dot-active\" style=\"left:0%;\"></span><span class=\"ant-slider-dot\" style=\"left:20%;\"></span><span class=\"ant-slider-dot\" style=\"left:40%;\"></span><span class=\"ant-slider-dot\" style=\"left:60%;\"></span><span class=\"ant-slider-dot\" style=\"left:80%;\"></span><span class=\"ant-slider-dot\" style=\"left:100%;\"></span></div><div class=\"ant-slider-handle\" style=\"left:0%;\"></div><div class=\"ant-slider-mark\"><span class=\"ant-slider-mark-text ant-slider-mark-text-active\" style=\"width:18%;margin-left:-9%;left:0%;\">Early-stage DevOps</span><span class=\"ant-slider-mark-text\" style=\"width:18%;margin-left:-9%;left:20%;\"></span><span class=\"ant-slider-mark-text\" style=\"width:18%;margin-left:-9%;left:40%;\"></span><span class=\"ant-slider-mark-text\" style=\"width:18%;margin-left:-9%;left:60%;\"></span><span class=\"ant-slider-mark-text\" style=\"width:18%;margin-left:-9%;left:80%;\"></span><span class=\"ant-slider-mark-text\" style=\"width:18%;margin-left:-9%;left:100%;\"><strong>DevOps at scale</strong></span></div></div></div></div><br/><br/> <br/><div class=\"ant-row\"><div class=\"ant-col-10\"><h3>Select a stage: </h3></div><div class=\"ant-col-2\"></div><div class=\"ant-col-12\"><div class=\"ant-radio-group\"><label class=\"ant-radio-wrapper\" style=\"display:block;height:30px;line-height:30px;\"><span class=\"ant-radio\"><input type=\"radio\" class=\"ant-radio-input\"/><span class=\"ant-radio-inner\"></span></span><span>People</span></label><label class=\"ant-radio-wrapper\" style=\"display:block;height:30px;line-height:30px;\"><span class=\"ant-radio\"><input type=\"radio\" class=\"ant-radio-input\"/><span class=\"ant-radio-inner\"></span></span><span>Process</span></label><label class=\"ant-radio-wrapper\" style=\"display:block;height:30px;line-height:30px;\"><span class=\"ant-radio\"><input type=\"radio\" class=\"ant-radio-input\"/><span class=\"ant-radio-inner\"></span></span><span>Technology</span></label><label class=\"ant-radio-wrapper\" style=\"display:block;height:30px;line-height:30px;\"><span class=\"ant-radio\"><input type=\"radio\" class=\"ant-radio-input\"/><span class=\"ant-radio-inner\"></span></span><span>Results</span></label></div></div></div></div><br/> <br/> <br/><div><button id=\"submit\" type=\"button\" class=\"ant-btn ant-btn-primary\"><span>Submit</span></button></div></div><div style=\"margin:100px;\"></div></div>"`;
